File: DemoApp/DemoApp/VideoDetailViewController.m
//
//  VideoDetailViewController.m
//  DemoApp
//
//  Created by Glispa GmbH on 04/12/15.
//  Copyright © 2015 Glispa GmbH All rights reserved.
//

#import "VideoDetailViewController.h"
#import "AppControllerManagement.h"


@interface VideoDetailViewController ()

@property(strong, nonatomic) AVOVideo *video;

@end

@implementation VideoDetailViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

#pragma mark - Actions

- (IBAction)loadClicked:(id)sender {
    __weak typeof(self) weakSelf = self;
    [AvocarrotSDK.sharedSDK loadVideoWithAdUnitId:self.adUnitId
                                          success:^(AVOVideo *video) {
                                              __strong __typeof__(self) sSelf = weakSelf;
                                              sSelf.video = video;
                                              NSLog(@"Video has loaded!");
                                          }
                                          failure:^(AVOError * _Nonnull error) {
                                              NSLog(@"Video loading error: %@", [error localizedDescription]);
                                          }];
}

- (IBAction)showClicked:(id)sender {
    [self.video showFromViewController:self];
}

- (IBAction)loadAndShowClicked:(id)sender {
    __weak typeof(self) weakSelf = self;
    [AvocarrotSDK.sharedSDK loadVideoWithAdUnitId:self.adUnitId
                                          success:^(AVOVideo *video) {
                                               __strong __typeof__(self) sSelf = weakSelf;
                                              [video showFromViewController:sSelf];
                                              NSLog(@"Video has loaded! And it will be shown...");
                                          }
                                          failure:^(AVOError * _Nonnull error) {
                                               NSLog(@"Video loading error: %@", [error localizedDescription]);
                                           }];
}

@end


@interface VideoDetailViewController (AppManagement) <AppControllerManagement>
@end

@implementation VideoDetailViewController (AppManagement)

- (void)applyUserData:(NSDictionary <NSString *, NSObject *> *)userData {
    if ([userData[@"adUnitId"] isKindOfClass:[NSString class]]) {
        self.adUnitId = (NSString *)userData[@"adUnitId"];
    }
}

@end

File: DemoApp/DemoAppSwift/VideoDetailViewController.swift
//
//  VideoDetailViewController.swift
//  DemoApp
//
//  Created by Glispa GmbH on 19/05/16.
//  Copyright © 2016 Glispa GmbH. All rights reserved.
//

import UIKit
import AvocarrotVideo

class VideoDetailViewController: UIViewController {

    @IBOutlet weak var loadButton: UIButton!
    @IBOutlet weak var showButton: UIButton!
    @IBOutlet weak var loadAndShowButton: UIButton!

    public var adUnitId: String!

    private var video: AVOVideo?

    override func viewDidLoad() {
        super.viewDidLoad()
        showButton.isEnabled = false
    }

    // MARK: - Actions

    @IBAction func loadClicked(_ sender: UIButton) {
        showButton.isEnabled = false
        AvocarrotSDK.shared.loadVideo(withAdUnitId: adUnitId, success: { [weak self] (video) in
                                            guard let sSelf = self else {return}
                                            sSelf.video = video
                                            sSelf.showButton.isEnabled = true
                                            print("Video has loaded!")
                                        }, failure: { (error) in
                                            print("Video loading error: \(error.localizedDescription)")
                                        })
    }

    @IBAction func showClicked(_ sender: UIButton) {
        self.video?.show(from: self)
        showButton.isEnabled = false
    }

    @IBAction func loadAndShowClicked(_ sender: UIButton) {
        showButton.isEnabled = false
        AvocarrotSDK.shared.loadVideo(withAdUnitId: adUnitId, success: { [weak self] (video) in
                                        guard let sSelf = self else {return}
                                        sSelf.video = video
                                        video.show(from: sSelf)
                                        print("Video has loaded! And it will be shown...")
                                    }, failure: { (error) in
                                        print("Video loading error: \(error.localizedDescription)")
                                    })
    }

}

extension VideoDetailViewController: AppControllerManagement {
    func applyUserData(_ userData: [String:Any]) {
        if let adId = userData["adUnitId"] as? String {
            adUnitId = adId
        }
    }
}
